blueflood-elasticsearch/src/main/java/com/rackspacecloud/blueflood/io/ElasticIO.java

```diff
@@ -35,39 +32,57 @@
 import org.slf4j.LoggerFactory;
 
 import java.io.IOException;
-import java.util.*;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
 
 public class ElasticIO extends AbstractElasticIO {
 
     public static final String ES_DOCUMENT_TYPE = "metrics";
 
-    private static final Logger log = LoggerFactory.getLogger(DiscoveryIO.class);;
+    private static final Logger logger = LoggerFactory.getLogger(ElasticIO.class);
 
+    //TODO: Refactor candidate?
     public ElasticIO() {
         this(RemoteElasticSearchServer.getInstance());
     }
 
+    //TODO: Refactor candidate?
     public ElasticIO(Client client) {
         this.client = client;
     }
 
+    //TODO: Refactor candidate?
     public ElasticIO(ElasticClientManager manager) {
         this(manager.getClient());
     }
 
 
+    /**
+     * Insert single metric document.
+     * @param metric
+     * @throws IOException
+     */
     public void insertDiscovery(IMetric metric) throws IOException {
-        List<IMetric> batch = new ArrayList<IMetric>();
+        //TODO: why there is no input validation for metric parameter? What will happen when it's null?
```

Comment:
It seems reasonable to add the guard statement then.

Reply:
I just added "TODO"s to make sure, we have some flags to look at when we are coming back to this. As I was already going through the code, I wanted to call these out, so that I don't have to rework on this, and take things from the TODOs and fix those whenever I am coming back to this code.